test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -117,304 +120,380 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
-			BOOST_CHECK_MESSAGE(lastTrueBlockHash == o["lastblockhash"].get_str(),
-					testname + "Boost check: lastblockhash does not match " + lastTrueBlockHash + " expected: " + o["lastblockhash"].get_str());
-
-			//Check final state (just to be sure)
-			BOOST_CHECK_MESSAGE(toString(testChain.topBlock().state().rootHash()) ==
-								toString(blockchain.topBlock().state().rootHash()),
-								testname + "State root in chain from RLP blocks != State root in chain from Field blocks!");
-
-			State postState(State::Null); //Compare post states
-			BOOST_REQUIRE((o.count("postState") > 0));
-			ImportTest::importState(o["postState"].get_obj(), postState);
-			ImportTest::compareStates(postState, testChain.topBlock().state());
-			ImportTest::compareStates(postState, blockchain.topBlock().state());
+	TestBlockChain testChain(genesisBlock);
+	assert(testChain.interface().isKnown(genesisBlock.blockHeader().hash(WithSeal)));
+
+	if (_o.count("genesisRLP") > 0)
+	{
+		TestBlock genesisFromRLP(_o["genesisRLP"].get_str());
+		checkBlocks(genesisBlock, genesisFromRLP, testname);
+	}
+
+	for (auto const& bl: _o["blocks"].get_array())
+	{
+		mObject blObj = bl.get_obj();
+		TestBlock blockFromRlp;
+		try
+		{
+			TestBlock blRlp(blObj["rlp"].get_str());
+			blockFromRlp = blRlp;
+			if (blObj.count("blockHeader") == 0)
+				blockFromRlp.noteDirty();			//disable blockHeader check in TestBlock
+			testChain.addBlock(blockFromRlp);
+		}
+		// if exception is thrown, RLP is invalid and no blockHeader, Transaction list, or Uncle list should be given
+		catch (Exception const& _e)
+		{
+			cnote << testname + "state sync or block import did throw an exception: " << diagnostic_information(_e);
+			checkJsonSectionForInvalidBlock(blObj);
+			continue;
+		}
+		catch (std::exception const& _e)
+		{
+			cnote << testname + "state sync or block import did throw an exception: " << _e.what();
+			checkJsonSectionForInvalidBlock(blObj);
+			continue;
+		}
+		catch (...)
+		{
+			cnote << testname + "state sync or block import did throw an exception\n";
+			checkJsonSectionForInvalidBlock(blObj);
+			continue;
+		}
+
+		//block from RLP successfully imported. now compare this rlp to test sections
+		BOOST_REQUIRE(blObj.count("blockHeader"));
+
+		//Check Provided Header against block in RLP
+		mObject emptyState;
+		TestBlock blockFromFields(blObj["blockHeader"].get_obj(), emptyState);
+		//ImportTransactions
+		BOOST_REQUIRE(blObj.count("transactions"));
+		for (auto const& txObj: blObj["transactions"].get_array())
+		{
+			TestTransaction transaction(txObj.get_obj());
+			blockFromFields.addTransaction(transaction);
 		}
-	}//for tests
+
+		// ImportUncles
+		if (blObj["uncleHeaders"].type() != json_spirit::null_type)
+			for (auto const& uBlHeaderObj: blObj["uncleHeaders"].get_array())
+			{
+				mObject uBlH = uBlHeaderObj.get_obj();
+				BOOST_REQUIRE((uBlH.size() == 16));
+
+				mObject emptyState;
```

Comment:
A variable of the same name already exists.  Maybe you can reuse the previously defined `emptyState`.